verify-push-backend/functions/challenge-status.js

```diff
@@ -0,0 +1,83 @@
+/**
+ *  Fetch the status of a challenge
+ *
+ *  Shows whether a challenge has been approved or denied. Can poll this endpoint once a challenge has been issued.
+ *
+ *
+ *  Pre-requisites
+ *  - Create a Verify Service (https://www.twilio.com/console/verify/services)
+ *  - Create a Challenge (https://www.twilio.com/docs/verify/api/challenge)
+ *
+ *  Parameters
+ *  - identity - required
+ *  - sid - required - challenge SID
+ *
+ *
+ *  Returns JSON
+ *
+ *  on Success:
+ *  challenge JSON - https://www.twilio.com/docs/verify/api/challenge#challenge-properties
+ *
+ *  on Error:
+ *  {
+ *    "error" {
+ *       "message": "Details about your error",
+ *       "moreInfo": "Link to error"
+ *    }
+ *  }
+ */
+
+function requiredParameter(param, paramName, response, callback) {
+  if (typeof param === "undefined") {
+    response.setBody({
+      error: {
+        message: `Missing parameter; please provide a '${paramName}'.`,
+        moreInfo:
+          "https://www.twilio.com/docs/verify/api/challenge#create-a-challenge-resource",
+      },
+    });
+    response.setStatusCode(400);
+    return callback(null, response);
```

Comment:
While this code will work as expected, it may look confusing to someone who doesn't understand how Twilio Functions code is executed, particularly that the callback is the end of the function.

I might suggest something more like this:

```js
function detectMissingParams(paramNames, event) {
  return paramNames.reduce((acc, param) => {
    if (typeof event[param] === "undefined") {
      acc.push(param);
    }
    return acc;
  }, []);
}
```

`detectMissingParams` takes a list of parameter names and the event and returns an array of missing parameters. You can then use it in the body of the function like:

```js
const missingParams = detectMissingParams(["identity", "sid"], event);
if (missingParams.length > 0) {
  response.setStatusCode(400);
  response.setBody({
    error: {
      message: `Missing parameter; please provide: '${missingParams.join(", ")}'.`,
      moreInfo: "https://www.twilio.com/docs/verify/api/challenge#create-a-challenge-resource",
    },
  });
  return callback(null, response);
} 
```

In this case, the error message will also contain all the missing params, not just the first one that was detected missing.

Reply:
good call, I like it!